root/meta/clingTErrorDiagnostics.ref

```diff
@@ -0,0 +1,11 @@
+
+input_line_26:1:12: error: non-void function 'f1' should return a value [-Wreturn-type]
```

Comment:
That seems to be very fragile as it's sensitive to context. Would this work?
```c++
gInterpreter->ProcessLine("#line 1 \"Test1\" \n int f1() { return; }");
```

Reply:
The ctest infrastructure seems to be filtering line numbers -as opposed to file names- before the diff (see below).
```
1189: -- BEGIN OUTDIFF OUTPUT --
1189: --- /home/jalopezg/Desktop/roottest/root/meta/clingTErrorDiagnostics.ref  Wed Sep  8 13:40:50 2021
1189: +++ /home/jalopezg/Desktop/build/roottest/root/meta/clingTErrorDiagnostics.log    Wed Sep  8 13:40:52 2021
1189: @@ -1,5 +1,5 @@
1189:
1189: -input_line:--:--:--:error:non-voidfunction'f1'shouldreturnavalue[-Wreturn-type]
1189: +Test1:--:--:error:non-voidfunction'f1'shouldreturnavalue[-Wreturn-type]
1189:  intf1(){return;}
1189:  ^
1189:  TestErrHdlr:3000:input_line:--:--:--:error:non-voidfunction'f2'shouldreturnavalue[-Wreturn-type]
1189:
1189: -- END OUTDIFF OUTPUT --
```

In general, this means that the `.ref` file is good enough, but that in any case it must match the expected output of the included `.C` file.